### ISSUE TITLE
Navigation overlay: "show arrow" options overly complex and confusing

### ISSUE THREAD

**Issue by davewhitley** [WordPress/gutenberg#75701]
## What problem does this address?

It is currently frustrating and confusing to use and edit the Navigation block settings related to the Submenu arrow.

As a user, it's hard to understand what conditions must be true to either **enable** or **hide** the "arrow" for submenus.

Currently:

1. If I want to require visitors to `click` on a menu to show the submenu I can't hide the arrow. Why? (I know the actual reason). I see no explanation for this in the UI. Disabling UI without explaining why (if it's not obvious) is a UX concern.
2. By not allowing the site builder to hide the arrow for `click` we are encouraging them to pick `hover` instead which is a worse UX for navigating submenus with a mouse.

Current UI:

<img width="283" height="198" alt="Image" src="https://github.com/user-attachments/assets/a52cdc84-87cf-47d7-a129-823e836fa172" />

<img width="283" height="385" alt="Image" src="https://github.com/user-attachments/assets/0551e51f-c168-4fa6-aa6e-4e83684c8a93" />

<img width="282" height="196" alt="Image" src="https://github.com/user-attachments/assets/03df8119-0724-4606-884f-562a6d75d206" />

## What is your proposed solution?

To solve problem number 1 above — let the site builder have the choice to disable the arrow, but show the a11y message so that the site builder can make an informed decision (just like the "hover" setting). I believe Hover and Click should be treated equally in this case. It would cause a lot less confusion.

The site builder may have accounted for a11y in other ways (add a unicode icon to the text, use the word "dropdown" in the menu, etc.), so I don't think it's good to assume that they shouldn't hide the arrow. And hopefully we'll be able to change the icon in the future with an icon block.
